lldb/source/Core/Module.cpp

```diff
@@ -1156,6 +1156,64 @@ bool Module::FileHasChanged() const {
   return m_file_has_changed;
 }
 
+void Module::ReportWarningOptimization(
+    llvm::Optional<lldb::user_id_t> debugger_id) {
+  ConstString file_name = GetFileSpec().GetFilename();
+  if (file_name.IsEmpty())
+    return;
+
+  StreamString ss;
+  ss << file_name.GetStringRef()
+     << " was compiled with optimization - stepping may behave "
+        "oddly; variables may not be available.";
+  Debugger::ReportWarning(std::string(ss.GetString()), debugger_id,
+                          &m_optimization_warning);
+}
+
+void Module::ReportWarningUnsupportedLanguage(
+    LanguageType language, llvm::Optional<lldb::user_id_t> debugger_id) {
+  StreamString ss;
+  ss << "This version of LLDB has no plugin for the language \""
+     << Language::GetNameForLanguageType(language)
+     << "\". "
+        "Inspection of frame variables will be limited.";
+  Debugger::ReportWarning(std::string(ss.GetString()), debugger_id,
+                          &m_language_warning);
+}
+
+void Module::ReportWarningCantLoadSwiftModule(
+    std::string details, llvm::Optional<lldb::user_id_t> debugger_id) {
+  StreamString ss;
+  ss << GetFileSpec().GetCString() << ": "
+     << "Cannot load Swift type information: " << details;
+  Debugger::ReportWarning(std::string(ss.GetString()), debugger_id,
+                          &m_swift_import_warning);
+}
+
+void Module::ReportWarningToolchainMismatch(
+    CompileUnit &comp_unit, llvm::Optional<lldb::user_id_t> debugger_id) {
+  if (SymbolFile *sym_file = GetSymbolFile()) {
+    llvm::VersionTuple sym_file_version =
+        sym_file->GetProducerVersion(comp_unit);
+    llvm::VersionTuple swift_version =
+        swift::version::Version::getCurrentCompilerVersion();
+    if (sym_file_version != swift_version) {
+      StreamString ss;
+      ss << GetFileSpec().GetFilename()
+         << " was compiled with a different Swift compiler "
+            "(version '"
+         << sym_file_version.getAsString()
+         << "') than the Swift compiler integrated into LLDB "
+            "(version '"
+         << swift_version.getAsString()
+         << "'). Swift expression evaluation requires a matching "
+            "compiler and debugger from the same toolchain.";
```

Comment:
This would probably read better, and grep better, expressed as a "format" string or printf.